### ISSUE TITLE
Create some kind of indicator showing the total number of open slots

### ISSUE THREAD

**Issue by Savathun** [DestinyItemManager/DIM#9540]
### Proposed change

I would like to see some kind of small little ui element that tells me the total number of open slots I have across my inventory. This should include open slots in the vault and across my characters, however this should *not* include any slots of a specific type on my characters that cannot be filled with items from the vault.

What I mean by that is, open slots on my characters in, for example, the class item slot, should not be counted if I have 10 or less class items for that specific class. I can give more specifics if that is not clear.

Additionally, the indicator should have a different color than usual if the number is only 2 (maybe orange) and another color if the number is less than 2 (red probably). 

### How does this fit into your workflow?

I am a player who is perpetually running close to a completely full vault and inventory on all 3 characters. Often, my vault is full or very near it, but I will have some open spaces on my characters. If I want to know how many spaces I have cleared up, I have to manually count from my characters inventories. Having a little indicator showing this number would mean I would no longer have to manually count if I want to know how many spaces I have open, which is useful knowledge before I decide to pull something from the post or start farming, as if I only have 2 slots total, that means I need to clear space before pulling otherwise I will run into problems.

**Comment by Savathun** [comment]
I would be happy to work on the change myself if this is approved but would probably need a few pointers on where to get started, I am not familiar with TS/JS or the DIM codebase.

**Comment by bhollis** [comment]
It's an interesting idea, though the implementation could be a bit tricky - as you pointed out, you'd need to compare the number of open slots in a bucket across all characters with the number of items in the vault that could actually be moved into that bucket. I wouldn't have rules that say you can only move items of the same class though - DIM doesn't follow that rule and it'd make the number appear lower than it should.

This would probably be best as a tooltip on the vault space count icon, showing extended details.

**Comment by Savathun** [comment]
Right. I edited that line to make that part more clear, that items of a specifc type should not be counted if there are less than 10*(number of chars) of them. 

pseudocode for calculating the number:

```
total_free_slots = spaces_free_in_vault
max_of_type = 10 * (number of characters)
for type in bucket_types:
   if total_items_of_type_not_in_post(type) > max_of_type:
       for character in characters:
           total_free_slots += free_slots_in_bucket(character, type)
```
